canopeum_frontend/src/components/analytics/site-modal/SiteForm.tsx

```diff
@@ -0,0 +1,270 @@
+use of register spreadability from 'react-hook-form'  */
+
+type Props = {
+  readonly availableSiteTypes: SiteType[],
+  readonly form: UseFormReturn<SiteFormDto>,
+  readonly formId: string,
+}
+
+const SiteForm = (
+  { availableSiteTypes, form, formId }: Props,
+) => {
+  const { t } = useTranslation()
+  const { translateValue } = useContext(LanguageContext)
+
+  const {
+    register,
+    getValues,
+    setValue,
+    watch,
+    control,
+    formState: { errors },
+  } = form
+
+  const siteImage = useWatch({ name: 'siteImage', control })
+  const [siteImageUrl, setSiteImageUrl] = useState<string>()
+
+  useEffect(() => {
+    if (siteImage instanceof File) {
+      setSiteImageUrl(URL.createObjectURL(siteImage))
+    }
+  }, [siteImage])
+
+  useEffect(() => {
+    const image = getValues('siteImage')
+    if (image instanceof File) {
+      setSiteImageUrl(URL.createObjectURL(image))
+    }
+
+    register('siteImage', {
+      required: t('analytics.site-modal.validation.image-required'),
+      validate: {
+        fileType: (file: File | undefined) =>
+          !file
+          || file.type.startsWith('image/')
+          || t('analytics.site-modal.validation.invalid-file-type'),
+      },
+    })
+
+    register('species', {
+      validate: species =>
+        species.length > 0
+        || t('analytics.site-modal.validation.tree-species-required'),
+    })
+  }, [register, watch, getValues])
+
+  return (
+    <form
+      className='d-flex flex-column gap-3'
+      id={formId}
+    >
+      <div>
+        <label aria-required className='form-label' htmlFor='site-name'>
+          {t('analytics.site-modal.site-name')}
+        </label>
+        <input
+          className='form-control'
+          id='site-name'
+          {...register('siteName', {
+            required: t('analytics.site-modal.validation.name-required'),
+          })}
+          type='text'
+        />
+        {errors.siteName && (
+          <span className='help-block text-danger'>{errors.siteName.message}</span>
+        )}
+      </div>
+
+      <div>
+        <label aria-required className='form-label' htmlFor='site-type'>
+          {t('analytics.site-modal.site-type')}
+        </label>
+        <select
+          className='form-select'
+          id='site-type'
+          {...register('siteType', {
+            required: t('analytics.site-modal.validation.site-type-required'),
+          })}
+        >
+          {availableSiteTypes.map(value => (
+            <option key={`available-specie-${value.id}`} value={value.id}>
+              {translateValue(value)}
+            </option>
+          ))}
+        </select>
+      </div>
+
+      <div>
+        <label aria-required className='form-label' htmlFor='site-image'>
+          {t('analytics.site-modal.site-image')}
+        </label>
+        <ImageUpload
+          id='site-image-upload'
+          imageUrl={siteImageUrl}
+          onChange={(file: File | null) => {
```

Comment:
By explicitly typing here, you're hiding that `file` cannot be `null`. Let TypeScript infer the type for you.

```suggestion
          onChange={file => {
```